core/sql/optimizer/BindItemExpr.cpp

```diff
@@ -6451,7 +6452,13 @@ NABoolean Like::beginEndKeysApplied(CollHeap *heap)
 
   return beginEndKeysApplied_;
 
-} // Like::beginEndKeysApplied()
+} // PatternMatchingFunction::beginEndKeysApplied()
+
+ItemExpr *Regexp::applyBeginEndKeys(BindWA *bindWA, ItemExpr *boundExpr,
+				  CollHeap *heap)
+{
+  return boundExpr;
```

Comment:
This is the place where we should add a check for non-default collating sequence, and raise error 4392 if found.